### PR TITLE
add ssd estimator

### PR DESCRIPTION
**Pull request by tmwangcas**

### CODE REVIEW

File: gluoncv/auto/estimators/ssd/ssd.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+from gluoncv.model_zoo import get_model
+from gluoncv.data.batchify import Tuple, Stack, Pad
+from gluoncv.data.transforms import presets
+from gluoncv.data.transforms.presets.ssd import SSDDefaultTrainTransform
+from gluoncv.data.transforms.presets.ssd import SSDDefaultValTransform
+from gluoncv.data.transforms.presets.ssd import SSDDALIPipeline
+from gluoncv.nn.bbox import BBoxClipToImage
+from gluoncv.utils.metrics.voc_detection import VOC07MApMetric
+from gluoncv.utils.metrics.coco_detection import COCODetectionMetric
+from gluoncv.utils.metrics.accuracy import Accuracy
```
Comment (chongruo) [inline gluoncv/auto/estimators/ssd/ssd.py:25]: should use relative imports as estimators/ is under the gluoncv folder.  Eg. from ...model_zoo import get_model

File: gluoncv/auto/estimators/ssd/utils.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+from gluoncv import data as gdata
+from gluoncv.data.batchify import Tuple, Stack, Pad
+from gluoncv.data.sampler import SplitSortedBucketSampler
+from gluoncv.data.transforms.presets.ssd import SSDDefaultTrainTransform
+from gluoncv.data.transforms.presets.ssd import SSDDefaultValTransform
+from gluoncv.data.transforms.presets.ssd import SSDDALIPipeline
+from gluoncv.utils.metrics.voc_detection import VOC07MApMetric
+from gluoncv.utils.metrics.coco_detection import COCODetectionMetric
+from gluoncv.utils.metrics.accuracy import Accuracy
```
Comment (chongruo) [inline gluoncv/auto/estimators/ssd/utils.py:15]: same here

File: gluoncv/auto/estimators/ssd/utils.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                                                     annotations_file=coco_annotations, device_id=i) for i, _ in enumerate(devices)]
+
+        # validation
+        if (not args.horovod or hvd.rank() == 0):
+            val_dataset = gdata.COCODetection(root=os.path.join(args.dataset_root + '/coco'),
+                                              splits='instances_val2017',
+                                              skip_empty=False)
+            val_metric = COCODetectionMetric(
+                val_dataset, os.path.join(args.logdir, args.save_prefix + '_eval'), cleanup=True,
+                data_shape=(args.data_shape, args.data_shape))
```
Comment (chongruo) [inline gluoncv/auto/estimators/ssd/utils.py:91]: args.ssd.data_shape

Comment (tmwangcas) [reply gluoncv/auto/estimators/ssd/utils.py]: Thx Chongruo, the problems have been corrected.

File: gluoncv/auto/estimators/ssd/ssd.py (hunk 1 above)
Comment (zhreshold) [reply gluoncv/auto/estimators/ssd/ssd.py]: we should use relative import here

File: gluoncv/auto/estimators/ssd/default.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+def valid_cfg():
+    # Epoch interval for validation
+    val_interval = 1
+
+
+ex = Experiment('ssd_default', ingredients=[logging, train, validation, ssd])
+
+
+@ex.config
+def default_configs():
```
Comment (zhreshold) [inline gluoncv/auto/estimators/ssd/default.py:70]: as discussed offline, we can use `get_ssd` function and expose more options to customize the network structure here

File: scripts/detection/ssd/fit_ssd.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+"""Train SSD"""
+
+import gluoncv as gcv
+
+gcv.utils.check_version('0.7.0')
```
Comment (zhreshold) [inline scripts/detection/ssd/fit_ssd.py:5]: 0.8.0

File: gluoncv/model_zoo/model_zoo.py [hunk 6]
```diff
@@ -127,6 +127,7 @@
     'ssd_300_mobilenet0.25_voc': ssd_300_mobilenet0_25_voc,
     'ssd_300_mobilenet0.25_coco': ssd_300_mobilenet0_25_coco,
     'ssd_300_mobilenet0.25_custom': ssd_300_mobilenet0_25_custom,
+    'custom_ssd': custom_ssd,
```
Comment (zhreshold) [inline gluoncv/model_zoo/model_zoo.py:130]: this is not necessary in model_zoo

**Pull request merged**